### ISSUE TITLE
Heading tags for accessibility

### ISSUE THREAD

**Issue by steverep** [home-assistant/home-assistant-polymer#3387]
**Home Assistant release with the issue:**

0.96.1

**Last working Home Assistant release (if known):**
None

**UI (States or Lovelace UI?):**

Lovelace

**Browser and Operating System:**

n/a

**Description of problem:**

Many pages throughout the UI either have headings that are not tagged or lack headings altogether.  I am visually impaired and efficient navigation with a screen reader relies on headings to quickly understand and jump focus around the page.  Please add tags for existing headings at an appropriate level and review where it would make sense to add others.  Please see the following WCAG techniques:
- [G141: Organizing a page using headings](https://www.w3.org/WAI/WCAG21/Techniques/general/G141)
- [H42: Using h1-h6 to identify headings](https://www.w3.org/WAI/WCAG21/Techniques/html/H42)

**Comment by steverep** [comment]
This one has been open for a year now, and it happens to be one of the simplest accessibility techniques to understand and implement:
- [ ] Give all card headings an `h2` and page titles an `h1`
- [ ] Give any modal dialog heading an `h1`
- [ ] Go through config pages and give out appropriate heading roles

Just applying appropriate tags to headings that are already there and visible would make a big difference to a screen reader user.

**Comment by NirmalKumarJoshi** [comment]
Hi @steverep please assign this to me. I would like to work on this

**Comment by mrand** [comment]
@NirmalKumarJoshi, one of the dev's may have to assign it to you

**Comment by Nardol** [comment]
Another approach could be to use heading ARIA roles.
I need to be more familiar with the frontend code but I made a pull request to add headings to integration list, which could be applied to other cards using the same way.
I'll try my best to do it but before asking to be assigned to this issue I prefer to be sure I could do it right.

Using the role instead of H1 etc avoid style issues and allows screen readers to get the heading information which would improve navigation as asked.
I use a screen reader and wish the same, headings would allow use to have a better navigation experience.

**Comment by AJolly** [comment]
the same issue applies to page titles.  For example, these pages all have the title of "Settings"
/config/integrations/dashboard
/config/devices/dashboard
/config/entities

**Comment by steverep** [comment]
@AJolly I agree, but can you open a separate issue for that and tag me?  It should be tracked separately as the fix is quite different.
